providers/implementations/keymgmt/ec_kmgmt.c

```diff
@@ -0,0 +1,592 @@
+                                          OSSL_PKEY_PARAM_EC_PUB_KEY,
+                                          pub_key, pub_key_len))
+        goto err;
+
+    if (priv_key != NULL) {
+        size_t sz;
+        int ecbits;
+        int ecdh_cofactor_mode = 0;
+
+        ecbits = EC_GROUP_order_bits(ecg);
+        sz = (ecbits + 7 ) / 8;
+        if (ecbits > 0) {
+            if (!ossl_param_bld_push_BN_pad(tmpl,
+                                            OSSL_PKEY_PARAM_EC_PRIV_KEY,
+                                            priv_key, sz))
+                goto err;
+        } else {
+            /*
+             * There are weird groups out there, so we can't error in this
+             * case, but all bets are off about constant-timeness.
+             */
+            if (!ossl_param_bld_push_BN(tmpl,
+                                        OSSL_PKEY_PARAM_EC_PRIV_KEY,
+                                        priv_key))
+                goto err;
+        }
+
+        ecdh_cofactor_mode =
+            (EC_KEY_get_flags(eckey) & EC_FLAG_COFACTOR_ECDH) ? 1 : 0;
+        if (!ossl_param_bld_push_int(tmpl,
+                                     OSSL_PKEY_PARAM_ECDH_COFACTOR_MODE,
+                                     ecdh_cofactor_mode))
+            goto err;
+    }
+
+    ret = 1;
+
+ err:
+    OPENSSL_free(pub_key);
+    return ret;
+}
+
+static
+void *ec_newdata(void *provctx)
+{
+    return EC_KEY_new();
+}
+
+static
+void ec_freedata(void *keydata)
+{
+    EC_KEY_free(keydata);
+}
+
+static
+int ec_has(void *keydata, int selection)
+{
+    EC_KEY *ec = keydata;
+    int ok = 0;
+
+    if ((selection & EC_POSSIBLE_SELECTIONS) != 0)
+        ok = 1;
+
+    if ((selection & OSSL_KEYMGMT_SELECT_PUBLIC_KEY) != 0)
+        ok = ok && (EC_KEY_get0_public_key(ec) != NULL);
+    if ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0)
+        ok = ok && (EC_KEY_get0_private_key(ec) != NULL);
+    if ((selection & OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS) != 0)
+        ok = ok && (EC_KEY_get0_group(ec) != NULL);
+
+    return ok;
+}
+
+static
+int ec_import(void *keydata, int selection, const OSSL_PARAM params[])
+{
+    EC_KEY *ec = keydata;
+    int ok = 0;
+
+    if (ec == NULL)
+        return 0;
+
+    /*
+     * In this implementation, we can export/import only keydata in the
+     * following combinations:
+     *   - domain parameters only
+     *   - public key with associated domain parameters
+     *   - private key with associated public key and domain parameters
+     *
+     * This means:
+     *   - domain parameters must always be requested
+     *   - private key must be requested alongside public key
+     */
+    if ((selection & OSSL_KEYMGMT_SELECT_DOMAIN_PARAMETERS) == 0)
+        return 0;
+    if ((selection & OSSL_KEYMGMT_SELECT_PRIVATE_KEY) != 0
+            && (selection & OSSL_KEYMGMT_SELECT_PUBLIC_KEY) == 0)
+        return 0;
+
+    if ((selection & EC_POSSIBLE_SELECTIONS) != 0)
```

Comment:
Considering all the checks above this, this check is no longer necessary, you can simply set `ok = 1` unconditionally

Reply:
Done